nixos/modules/virtualisation/azure-common.nix

```diff
@@ -13,36 +22,37 @@ in
     };
   };
 
-  imports = [
-    ../profiles/headless.nix
-    ./azure-agent.nix
-  ];
+  imports = [ ./azure-agent.nix ];
 
   config = {
     virtualisation.azure.agent.enable = true;
-
-    boot.kernelParams = [ "console=ttyS0" "earlyprintk=ttyS0" "rootdelay=300" "panic=1" "boot.panic_on_fail" ];
-    boot.initrd.kernelModules = [ "hv_vmbus" "hv_netvsc" "hv_utils" "hv_storvsc" ];
+    services.cloud-init.enable = true;
+    services.cloud-init.network.enable = true;
+    systemd.services.cloud-config.serviceConfig.Restart = "on-failure";
+
+    boot.kernelParams = [
+      "console=ttyS0"
+      "earlyprintk=ttyS0"
+      "rootdelay=300"
+      "panic=1"
+      "boot.panic_on_fail"
+    ];
+    boot.initrd.kernelModules = [
+      "hv_vmbus"
+      "hv_netvsc"
+      "hv_utils"
+      "hv_storvsc"
+    ];
```

Comment:
While at it, if you know, please add comments that explain why the above are right here.

I believe it was already asked in some past review, and should really be a code comment.

Reply:
I have added more comments in the latest iteration, covering kernel modules, accelerated networking and grub configurations. Please let me know your thoughts.